crates/nix_rs/src/version.rs

```diff
@@ -10,14 +10,14 @@ use tracing::instrument;
 use crate::command::{NixCmd, NixCmdError};
 
 /// Nix version as parsed from `nix --version`
-#[derive(Clone, Copy, PartialOrd, PartialEq, Eq, Debug, SerializeDisplay, DeserializeFromStr)]
+#[derive(Clone, Copy, Debug, SerializeDisplay, DeserializeFromStr)]
 pub struct NixVersion {
     /// Major version
     pub major: u32,
     /// Minor version
-    pub minor: u32,
+    pub minor: Option<u32>,
     /// Patch version
-    pub patch: u32,
+    pub patch: Option<u32>,
```

Comment:
This changes the semantics of this type. If you look at the type documenation, it says 

```
Nix version as parsed from `nix --version`
```

And `nix --version` always returns all 3 components - major, minor and patch.

---

It seems that the type here is being changed to fit something in `version_spec.hs` - yet `version.rs` is isolated from that module.

Reply:
Reverted changes to `version.rs`, the edge case of not having `minor`, `patch` or both `minor` and `patch` components, is now handled by `version_spec.rs`